Update mets_factory.py

kevdpow opened the pull request:
I encountered an error when processing directories on a samba share; ordered_file_list() returned an out-of-order list. I suggest returning a sorted output_file_list instead.

kevdpow committed:
Update mets_factory.py

I encountered an error when processing directories of files on a samba share; ordered_file_list() returned an out-of-order list. I suggest returning a sorted output_file_list instead.

JohnnyFalvo commented:
Hi Kevin,
That looks like a pretty smart change. I haven't seen this issue before; the rationale behind not using sorted() on non-numeric filenames was that we wanted the filesystem to make the call as to how they should be ordered, rather than making any naive assumptions about intended ordering (numerical filenames felt like a safe bet, however). I'm not too wedded to that principle though.
Just to be on the safe side, could you let us know how you are calling the mets_factory (ie directly, through mets_dnx, sip factory etc) so I can try to reproduce the issue?
Many thanks,
Sean.

SeanMoselyNLNZ commented:
Whoops, that last comment was from my personal GitHub account :p
Kevin: Looks like it's not to do with Samba, but is a general bug that I can reproduce, which is great!
If we apply sorted() on the line you suggest, it has an unfortunate side effect for purely numeric filenames (like 1.txt, 2.txt, etc) -- it re-sorts them so they will no longer be in numerical order. 
However, if we change line 333 from:
>files = str_file_names + int_file_names
to:
>files = sorted(str_file_names) + int_file_names
then we keep the numeric filenames ordered as well as solving the issue you've raised.
I'll add a "request changes" to your pull request (so you can make the tweak and you'll keep the pull against your name), then I'll add the tests and update the version point.
Thanks for looking into this -- much appreciated!

SeanMoselyNLNZ reviewed (changes requested):
Suggest we leave line 336 as is, but change line 333 to:
files = sorted(str_file_names) + int_file_names

SeanMoselyNLNZ committed:
Update mets_factory.py

Tweaked changes so that numerical filenames will not be re-sorted back to alphabetic order
